framework/decode/dx12_browse_consumer.h

```diff
@@ -0,0 +1,272 @@
+/*
+** Copyright (c) 2023 LunarG, Inc.
+**
+** Permission is hereby granted, free of charge, to any person obtaining a
+** copy of this software and associated documentation files (the "Software"),
+** to deal in the Software without restriction, including without limitation
+** the rights to use, copy, modify, merge, publish, distribute, sublicense,
+** and/or sell copies of the Software, and to permit persons to whom the
+** Software is furnished to do so, subject to the following conditions:
+**
+** The above copyright notice and this permission notice shall be included in
+** all copies or substantial portions of the Software.
+**
+** THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+** IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+** FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+** AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+** LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+** FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+** DEALINGS IN THE SOFTWARE.
+*/
+
+#ifndef GFXRECON_DECODE_DX12_BROWSE_CONSUMER_H
+#define GFXRECON_DECODE_DX12_BROWSE_CONSUMER_H
+
+#include "generated/generated_dx12_decoder.h"
+#include "graphics/dx12_dump_resources.h"
+
+GFXRECON_BEGIN_NAMESPACE(gfxrecon)
+GFXRECON_BEGIN_NAMESPACE(decode)
+
+struct TrackDumpCommandList
+{
+    uint64_t begin_renderpass_code_index{ 0 };
+    uint64_t set_render_targets_code_index{ 0 };
+    uint64_t drawcall_code_index{ 0 };
+    uint64_t execute_code_index{ 0 };
+
+    // vertex
+    std::vector<D3D12_GPU_VIRTUAL_ADDRESS> captured_vertex_buffer_view_gvas;
+
+    // index
+    D3D12_GPU_VIRTUAL_ADDRESS captured_index_buffer_view_gva{ decode::kNullGpuAddress };
+
+    // descriptor
+    std::vector<format::HandleId> descriptor_heap_ids;
+
+    // render target
+    // Track render target info in replay, not here.
+    // Because the useful info is replay cpuDescriptor. It's only available in replay.
+
+    void Clear()
+    {
+        begin_renderpass_code_index = 0;
+        drawcall_code_index         = 0;
+        execute_code_index          = 0;
+        captured_vertex_buffer_view_gvas.clear();
+        captured_index_buffer_view_gva = decode::kNullGpuAddress;
+        descriptor_heap_ids.clear();
+    }
+};
+
+class Dx12BrowseConsumer : public Dx12Consumer
```

Comment:
`Dx12BrowseConsumer` browses the file before replay to find the target drawcall code index and some key code indices, and track resource infos. That could help replay easier. But it doesn't track render target infos because the useful info is replay cpuDescriptor. It's only available in replay.